test/functional/core/fileio_spec.lua

```diff
@@ -128,5 +129,12 @@ describe('fileio', function()
     table.insert(text, '')
     eq(text, funcs.readfile(fname, 'b'))
   end)
+  it('read invalid u8 over INT_MAX doesn\'t segfault', function()
+    clear()
+    command('call writefile(0zFFFFFFFF, "Xtest-u8-int-max")')
+    local fname = 'Xtest-u8-int-max'
+    -- This should not segfault
+    command("edit ++enc=utf32 "..fname)
```

Comment:
You can use the `assert_alive()` helper (need to import it at the top level):

```suggestion
    -- This should not segfault
    command("edit ++enc=utf32 "..fname)
    assert_alive()
```